tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorService.cs

```diff
@@ -252,86 +229,62 @@ public CommandLineSuggestion GetSuggestion(PredictionContext context, int sugges
         }
 
         /// <inheritdoc/>
-        public virtual void RequestPredictions(IEnumerable<string> commands)
+        public virtual async Task<bool> RequestPredictionsAsync(IEnumerable<string> commands, CancellationToken cancellationToken)
         {
             Validation.CheckArgument(commands, $"{nameof(commands)} cannot be null.");
 
-            var localCommands= string.Join(AzPredictorConstants.CommandConcatenator, commands);
-            bool postSuccess = false;
-            Exception exception = null;
-            bool startRequestTask = false;
+            var localCommands = string.Join(AzPredictorConstants.CommandConcatenator, commands);
+            bool isRequestSent = false;
 
             try
             {
                 if (string.Equals(localCommands, _commandToRequestPrediction, StringComparison.Ordinal))
                 {
                     // It's the same history we've already requested the prediction for last time, skip it.
-                    return;
+                    return false;
                 }
 
                 if (commands.Any())
                 {
                     SetCommandToRequestPrediction(localCommands);
 
-                    // When it's called multiple times, we only need to keep the one for the latest command.
-
-                    _predictionRequestCancellationSource?.Cancel();
-                    _predictionRequestCancellationSource = new CancellationTokenSource();
+                    AzPredictorService.ReplaceThrottleUserIdToHeader(_client?.DefaultRequestHeaders, _azContext.UserId);
 
-                    var cancellationToken = _predictionRequestCancellationSource.Token;
-
-                    // We don't need to block on the task. We send the HTTP request and update prediction list at the background.
-                    startRequestTask = true;
-                    Task.Run(async () => {
-                        try
-                        {
-                            AzPredictorService.ReplaceThrottleUserIdToHeader(_client?.DefaultRequestHeaders, _azContext.UserId);
+                    var requestContext = new PredictionRequestBody.RequestContext()
+                    {
+                        SessionId = _telemetryClient.SessionId,
+                        CorrelationId = _telemetryClient.CorrelationId,
```

Comment:
Just a question, do we know how long this Correlation ID persists? I.e. is it the length of the login, a max of 20 minutes, etc

Reply:
It depends. The value is changed when a command is executed. That varies.